fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -370,13 +369,14 @@ public PlanFragment visitPhysicalHashJoin(PhysicalHashJoin<Plan, Plan> hashJoin,
         if (JoinUtils.shouldNestedLoopJoin(hashJoin)) {
             throw new RuntimeException("Physical hash join could not execute without equal join condition.");
         } else {
-            Expression eqJoinExpression = hashJoin.getCondition().get();
-            List<Expr> execEqConjunctList = ExpressionUtils.extractConjunction(eqJoinExpression).stream()
-                    .map(EqualTo.class::cast)
-                    .map(e -> swapEqualToForChildrenOrder(e, hashJoin.left().getOutput()))
-                    .map(e -> ExpressionTranslator.translate(e, context))
-                    .collect(Collectors.toList());
-
+            List<Expr> execEqConjunctList = new ArrayList<>();
+            if (hashJoin.getOnClauseCondition().isPresent()) {
+                execEqConjunctList = hashJoin.getHashJoinPredicates().stream()
+                        .map(EqualTo.class::cast)
+                        .map(e -> swapEqualToForChildrenOrder(e, hashJoin.left().getOutput()))
+                        .map(e -> ExpressionTranslator.translate(e, context))
+                        .collect(Collectors.toList());
+            }
```

Comment:
```suggestion
            hashJoin.getOnClauseCondition().orElse(Lists.newArrayList()).stream()
                    .map(EqualTo.class::cast)
                    .map(e -> swapEqualToForChildrenOrder(e, hashJoin.left().getOutput()))
                    .map(e -> ExpressionTranslator.translate(e, context))
                    .collect(Collectors.toList());
```

Reply:
don't apply directly.

Reply:
can we save the effort of running the map and collect on an empty list?